pkg/apis/v1beta1/nodepool.go

```diff
@@ -177,6 +177,7 @@ type ObjectMeta struct {
 
 // NodePool is the Schema for the NodePools API
 // +kubebuilder:object:root=true
+// +kubebuilder:storageversion
```

Comment:
We need to set  one of the version of the CRDs as a `storageversion`, which is used in setting a version is set as the "storage version" to be stored in `etcd`. The NodeClaim already for v1beta1. https://github.com/kubernetes-sigs/karpenter/blob/f2de2b4fa76c0ec8ac9115d9c0d62dae481c4ef6/pkg/apis/v1beta1/nodeclaim.go#L166

Reply:
Leaving the `storageVersion` on `v1beta1`, will avoid users having the to migrate their resources `storageversion` https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version

Reply:
I think this is wrong. We want the storage version to be v1. Else, when new things are applied, they are still going to be sticking on the old v1beta1 storage version. As long as that's the case, we can't drop v1beta1 from the set of versions. Effectively, we need all new resources to move to the v1 storage version, then drop the v1beta1 support alongside the webhook

Reply:
See: https://book.kubebuilder.io/reference/markers/crd#:~:text=kubebuilder%3Astorageversion

Reply:
The idea here was to make sure we were consistent between the NodePool and NodeClaim. NodeClaim had explicitly marked that v1beta1 would be the storage version. We will need to make sure to push customers to using the v1 APIs now then, so that they can easily migrate the CR version in etcd.